lib/spack/spack/build_systems/cmake.py

```diff
@@ -97,35 +103,34 @@ def build_directory(self):
     def cmake_args(self):
         """Method to be overridden. Should return an iterable containing
         all the arguments that must be passed to configure, except:
-        - CMAKE_INSTALL_PREFIX
-        - CMAKE_BUILD_TYPE
+        * CMAKE_INSTALL_PREFIX
+        * CMAKE_BUILD_TYPE
         """
         return []
 
     def cmake(self, spec, prefix):
         """Run cmake in the build directory"""
         options = [self.root_cmakelists_dir()] + self.std_cmake_args + \
             self.cmake_args()
-        create = not os.path.exists(self.build_directory())
-        with working_dir(self.build_directory(), create=create):
+        with working_dir(self.build_directory(), create=True):
```

Comment:
Looks like it calls `mkdirp`, which does indeed check to see whether or not the directory exists first, so we can set `create=True` and not worry.